2020-07-14 14:01:26,146 ERROR: Exception on /index [GET] [in /home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/austin/Desktop/LearnFlask/microblog/app/routes.py", line 29, in index
    return render_template('index.html', title='Home', form=form, posts=posts)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/base.html", line 32, in top-level template code
    {% block content %}{% endblock %}
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/index.html", line 20, in block "content"
    {% for post in posts %}
TypeError: 'Pagination' object is not iterable
2020-07-14 14:01:27,440 ERROR: Exception on /explore [GET] [in /home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/austin/Desktop/LearnFlask/microblog/app/routes.py", line 37, in explore
    return render_template('index.html', title='Explore', posts=posts)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/base.html", line 32, in top-level template code
    {% block content %}{% endblock %}
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/index.html", line 20, in block "content"
    {% for post in posts %}
TypeError: 'Pagination' object is not iterable
2020-07-14 14:01:52,052 ERROR: Exception on /explore [GET] [in /home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/austin/Desktop/LearnFlask/microblog/app/routes.py", line 37, in explore
    return render_template('index.html', title='Explore', posts=posts)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/home/austin/Desktop/LearnFlask/microblog/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/base.html", line 32, in top-level template code
    {% block content %}{% endblock %}
  File "/home/austin/Desktop/LearnFlask/microblog/app/templates/index.html", line 20, in block "content"
    {% for post in posts %}
TypeError: 'Pagination' object is not iterable
2020-07-14 14:06:54,818 INFO: Microblog startup [in /home/austin/Desktop/LearnFlask/microblog/app/__init__.py:30]
2020-07-14 14:09:25,378 INFO: Microblog startup [in /home/austin/Desktop/LearnFlask/microblog/app/__init__.py:30]
2020-07-14 14:11:16,589 INFO: Microblog startup [in /home/austin/Desktop/LearnFlask/microblog/app/__init__.py:30]
2020-07-14 20:28:21,963 INFO: Microblog startup [in /home/austin/Desktop/LearnFlask/microblog/app/__init__.py:32]
2020-07-14 20:32:42,562 INFO: Microblog startup [in /home/austin/Desktop/LearnFlask/microblog/app/__init__.py:45]
2020-07-14 20:36:43,409 INFO: Microblog startup [in /home/austin/Desktop/LearnFlask/microblog/app/__init__.py:45]
2020-07-15 09:31:19,634 INFO: Microblog startup [in /home/austin/Desktop/LearnFlask/microblog/app/__init__.py:45]
